youtube_dl/extractor/beampro.py

```diff
@@ -6,15 +6,17 @@
     ExtractorError,
     clean_html,
     compat_str,
+    float_or_none,
     int_or_none,
     parse_iso8601,
     try_get,
+    urljoin,
 )
 
 
 class BeamProLiveIE(InfoExtractor):
     IE_NAME = 'Beam:live'
-    _VALID_URL = r'https?://(?:\w+\.)?beam\.pro/(?P<id>[^/?#&]+)'
+    _VALID_URL = r'https?://(?:\w+\.)?beam\.pro/(?P<id>[^/?#&]+)$'
```

Comment:
The issue here is that `http://www.beam.pro/niterhayven?vod=123`, which is a VOD URL, would also get matched by this. I tried to fix this in ccc035c1fb2b6080b4dab9e2cb0b8273f734c1cc by reordering the import of those extractors, but I don't know if this acceptable.